Intermittent MultiFromNotTrustedInputStreamTckTest

[//]: # "Please remove these comments"
[//]: # "Provide a general summary of the issue in the Title above"

## PRs:

- [ ] 2x #5061
- [ ] 3x #3597
- [ ] 4x #5055


## Environment Details
pipeline

----------
```
default / build / io.helidon.common.reactive.MultiFromNotTrustedInputStreamTckTest.required_createPublisher3MustProduceAStreamOfExactly3Elements (from TestSuite)

Failing for the past 1 build (Since Unstable#2 )
Took 58 ms.
přidat popis
Error Message
Publisher org.reactivestreams.FlowAdapters$ReactivePublisherFromFlow@1174e173 produced only 2 elements expected [true] but found [false]
Stacktrace
java.lang.AssertionError: Publisher org.reactivestreams.FlowAdapters$ReactivePublisherFromFlow@1174e173 produced only 2 elements expected [true] but found [false]
	at org.testng.Assert.fail(Assert.java:96)
	at org.testng.Assert.failNotEquals(Assert.java:776)
	at org.testng.Assert.assertTrue(Assert.java:44)
	at org.reactivestreams.tck.PublisherVerification$2.run(PublisherVerification.java:185)
	at org.reactivestreams.tck.PublisherVerification.activePublisherTest(PublisherVerification.java:1142)
	at org.reactivestreams.tck.PublisherVerification.required_createPublisher3MustProduceAStreamOfExactly3Elements(PublisherVerification.java:179)
	at jdk.internal.reflect.GeneratedMethodAccessor18.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:571)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:707)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:979)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1187)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1116)
	at org.testng.TestNG.runSuites(TestNG.java:1028)
	at org.testng.TestNG.run(TestNG.java:996)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:136)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeMulti(TestNGDirectoryTestSuite.java:193)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:94)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:145)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
Standard Output
Configuring TestNG with: TestNG60Configurator
```

--- a/common/reactive/src/main/java/io/helidon/common/reactive/MultiFromBlockingInputStream.java
+++ b/common/reactive/src/main/java/io/helidon/common/reactive/MultiFromBlockingInputStream.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020 Oracle and/or its affiliates.
+ * Copyright (c) 2020, 2022 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -22,8 +22,8 @@ import java.nio.ByteBuffer;
 import java.util.Objects;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Flow;
-import java.util.concurrent.LinkedBlockingQueue;
-import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
 import java.util.function.IntSupplier;
 
 class MultiFromBlockingInputStream extends MultiFromInputStream {
@@ -68,9 +68,8 @@ class MultiFromBlockingInputStream extends MultiFromInputStream {
     static final class InputStreamSubscription extends MultiFromInputStream.InputStreamSubscription {
 
         private final ExecutorService executorService;
-        private final LinkedBlockingQueue<Runnable> submitQueue = new LinkedBlockingQueue<>();
+        private Lock lck = new ReentrantLock();
 
-        private final AtomicBoolean draining = new AtomicBoolean(false);
 
         InputStreamSubscription(Flow.Subscriber<? super ByteBuffer> downstream,
                                 InputStream inputStream,
@@ -81,24 +80,14 @@ class MultiFromBlockingInputStream extends MultiFromInputStream {
         }
 
         protected void trySubmit(long n) {
-            submitQueue.add(() -> {
-                submit(n);
-                drainSubmitQueue();
-            });
-            drainSubmitQueue();
-        }
-
-        private void drainSubmitQueue() {
-            if (!draining.getAndSet(true)) {
+            executorService.submit(() -> {
                 try {
-                    Runnable job = submitQueue.poll();
-                    if (job != null) {
-                        executorService.submit(job);
-                    }
+                    lck.lock();
+                    submit(n);
                 } finally {
-                    draining.set(false);
+                    lck.unlock();
                 }
-            }
+            });
         }
     }
 }

--- a/tests/tck/tck-reactive-streams/src/test/java/io/helidon/common/reactive/MultiFromNotTrustedInputStreamTckTest.java
+++ b/tests/tck/tck-reactive-streams/src/test/java/io/helidon/common/reactive/MultiFromNotTrustedInputStreamTckTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, 2021 Oracle and/or its affiliates.
+ * Copyright (c) 2020, 2022 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -61,5 +61,4 @@ public class MultiFromNotTrustedInputStreamTckTest extends MultiFromTrustedInput
                 .byteBufferSize(BUFFER_SIZE)
                 .build();
     }
-
 }

--- a/tests/tck/tck-reactive-streams/src/test/java/io/helidon/common/reactive/MultiFromTrustedInputStreamTckTest.java
+++ b/tests/tck/tck-reactive-streams/src/test/java/io/helidon/common/reactive/MultiFromTrustedInputStreamTckTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021 Oracle and/or its affiliates.
+ * Copyright (c) 2021, 2022 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -31,7 +31,7 @@ public class MultiFromTrustedInputStreamTckTest extends FlowPublisherVerificatio
     static final int BUFFER_SIZE = 4;
 
     public MultiFromTrustedInputStreamTckTest() {
-        super(new TestEnvironment(50));
+        super(new TestEnvironment(500));
     }
 
     @Override

